lib/block-supports/block-style-variations.php

```diff
@@ -0,0 +1,353 @@
+<?php
+/**
+ * Block support to enable per-section styling of block types via
+ * block style variations.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Get the class name for this application of this block's variation styles.
+ *
+ * @param array  $block     Block object.
+ * @param string $variation Slug for the block style variation.
+ *
+ * @return string The unique class name.
+ */
```

Comment:
I guess we'll need `@since` on all of these prior to merging 😬

Reply:
Done: https://github.com/WordPress/gutenberg/pull/57908/commits/33830c4a628890b8d3f6233356aff64a77ebfc90